lib/valueflow.cpp

```diff
@@ -1116,11 +1116,7 @@ static size_t accumulateStructMembers(const Scope* scope, F f)
         if (const ValueType* vt = var.valueType()) {
             if (vt->type == ValueType::Type::RECORD && vt->typeScope == scope)
                 return 0;
-            total = f(total, *vt);
-            const MathLib::bigint dim = std::accumulate(var.dimensions().cbegin(), var.dimensions().cend(), 1LL, [](MathLib::bigint i1, const Dimension& dim) {
-                return i1 * dim.num;
-            });
-            total *= dim;
+            total = f(total, *vt, var);
```

Comment:
Instead of passing the var, you could pass the `dim`. And then it can multiply it by the size.